ee/api/role.py

```diff
@@ -0,0 +1,110 @@
+from typing import cast
+
+from django.db import IntegrityError
+from rest_framework import mixins, serializers, viewsets
+from rest_framework.permissions import SAFE_METHODS, BasePermission, IsAuthenticated
+
+from ee.models.role import Role, RoleMembership
+from posthog.api.routing import StructuredViewSetMixin
+from posthog.api.shared import UserBasicSerializer
+from posthog.models import OrganizationMembership
+from posthog.models.user import User
+from posthog.permissions import OrganizationMemberPermissions, extract_organization
+
+DEFAULT_ROLE_NAME = "Write"
```

Comment:
Based on the mocks, I think admins can change the default role, right? (they'll need to make it 'read only' ).

I'm unclear on how this will change the model creation code below, since we can't imply permissions based on the role name 😅

Reply:
I think a way to do "read only" is by removing the write role, so an empty array of role permissions = view only. We don't have to add a `read` role but we can if it solves a problem that simply removing "write" doesn't?

Reply:
I see, so when an admin chooses to set default to 'read only', we remove every user in the org from `RoleMembership` to the `Write` role?

Reply:
yepp if by default everyone is "read only" then we delete the write role memberships for everyone. And so team member A will have empty role memberships until new roles are created and they're added to it. And if the admin wants to change it back to default "write access" then we add that "write" role membership back to everyone.

Reply:
So then we only have to check if any role membership exists or not for a user to determine "view only" mode for them. If a user doesn't have any role memberships then we don't even need to check for feature flags access for them !

Reply:
This is a bit problematic from a DB design perspective: https://en.wikipedia.org/wiki/Database_normalization

In particular, (yes I had to read that link again to come up with the right terms for things), there seems to be a transitive dependency with the role & permissions.

Disregard the default case for now since that might change, but consider a role like: `Schrodinger`, and every team member is a part of this role.

Every day, a cat opens up PostHog and walks on the keyboard, such that the permissions for this role toggle from `full write access` to `view only`. One day it's `view only`, the next day it's `full write access`, and then back, and so on.

Based on the above design, when we switch to `view only` we update `RoleMembership` to remove all users from the role. And same for when we switch to `full write access`, we add them back.

1. This in unnecessary updates to the DB, since changing permissions shouldn't affect `RoleMembership`, which can lead to issues with DB management, and hard to get right (since all rows to be updated need to be in an atomic transaction, etc. etc)
2. How do we figure out who was in the role, when permissions change? How do we add back only the people who had this role?

Reply:
1. Wouldn't updating all users to have a "read only" role by patching their role membership also require an atomic transaction? 
2. This makes sense I think? Can you explain how does having a default role that you can change the name of answer these problems? 

What do you suggest to do here instead?